### PR TITLE
Add test plan for QUARKUS-3456

### PR DESCRIPTION
**Pull request by michalvavrik**
### Links

JIRA: https://issues.redhat.com/browse/QUARKUS-3456

Quarkus documentation: https://quarkus.io/version/main/guides/config-secrets#protect-the-keystore-password

### Reminder for considerable topics

 - [x] Make sure you have considered the following areas when preparing the test plan:

   - Logging
   - Tracing
   - Metrics
   - Security
   - OpenAPI
   - Data sources
   - Frontend
   - Qute

### CODE REVIEW

File: QUARKUS-3456.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Three new integration test classes will be added to Quarkus CLI module of the Quarkus QE Test Suite.
+One for each `quarkus config` subcommand (`set`, `remove`, `encrypt`).
+Tests will run in a JVM mode only as intention is to test CLI and assure that `application.properties` are updated correctly. 
+
+## Impact on resources
+We will start JVM application once per each test class and call an endpoint or two, therefore additional test execution time should be below 30 seconds.
+
+## Future considerations
+
+- use more Ciphers (use different algorithms, modes and padding) to encrypt a secret
```
Comment (rsvoboda) [inline QUARKUS-3456.md:51]: I would prefer to move it out of the future scope and put it into current TD.
What would be the estimate to implement this?

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Few things to handle + we need confirmation of scope from https://issues.redhat.com/browse/QUARKUS-3456?focusedId=25160816&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25160816

File: QUARKUS-3456.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Test coverage must include:
+
+- `quarkus config set/remove/encrypt` command tested with options combined in a fashion customers are likely to use,
+- documented flow to create a secret, add it to the Keystore, set Keystore config source and check Quarks application decodes secrets,
+- default secret handler `AES/GCM/NoPadding` is used to create the secret,
+- secret is decoded in FIPS-enabled environment,
+- read encrypted configuration in Quarkus application user code,
+- store secrets generated with the CLI in a Keystore and access them using Quarkus application,
+- decode the encryption key in Base64, if the plain text key was used to encrypt the secret.
+
```
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-3456.md:28]: This is not a unit test, I cannot test negative secret handlers unless in a DEV mode, but this sounds like a job for upstream test coverage. Invalid secret handler must result in a Quarkus app build error, which is subject for upstream test coveage, not IT. We write ITs.

File: QUARKUS-3456.md (hunk 1 above)
Comment (michalvavrik) [reply QUARKUS-3456.md:51]: You can chose alg/mode/padding via CLI encrypt cmd options https://github.com/quarkusio/quarkus/blob/main/devtools/cli/src/main/java/io/quarkus/cli/config/Encrypt.java#L59 but I am worried about this https://github.com/quarkusio/quarkus/blob/main/devtools/cli/src/main/java/io/quarkus/cli/config/Encrypt.java#L64. I simply didn't get to test this and I don't know if you can use different cipher while you still have there GCM spec 128 and sha256.

I cannot tell if it is additional time or not, but https://github.com/quarkusio/quarkus/pull/34493 said `Can encrypt arbitrary values in AES/GCM/NoPadding (will support more algorithms)` so I didn't expect it is actually supported. Right now, it is not documented which other ciphers are supported.

File: QUARKUS-3456.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+Now Quarkus leverages Quarkus CLI and allows to create, update and remove configuration properties with it.
+One of options is to pass the CLI plain secret and Quarkus will encode this secret for you.
+You can even ask Quarkus CLI to set encoded secret directly to the application properties.
+Or, if you use YAML or other formats (like if you keep secrets in the environment variables), you can just copy it from the CLI output.
+We should test this new feature, including other sub-commands under `quarkus config` command as users are likely to use them when they encrypt secrets.
+
+## Scope of the testing
+Test coverage must include:
+
+- `quarkus config set/remove/encrypt` command tested with options combined in a fashion customers are likely to use,
```
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-3456.md:21]: I'll add it.

File: QUARKUS-3456.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-3456.md:28]: If you think it is necessary, I'll just do `start`, check it failed, change config properties, start and do testing. I think it shouldn't cost too much of a time

[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-3456.md:31]: done

File: QUARKUS-3456.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-3456.md:21]: done

**Comment by michalvavrik** [comment]
> we need confirmation of scope from

in a case scope wouldn't be confirmed, this would be blocked till TD is merged, I prefer to update TP later

File: QUARKUS-3456.md (hunk 1 above)
Comment (michalvavrik) [reply QUARKUS-3456.md:54]: @rsvoboda 
```
    @Option(hidden = true, names = { "-a", "--algorithm" }, description = "Algorithm", defaultValue = "AES")
    String algorithm;

    @Option(hidden = true, names = { "-m", "--mode" }, description = "Mode", defaultValue = "GCM")
    String mode;

    @Option(hidden = true, names = { "-p", "--padding" }, description = "Padding", defaultValue = "NoPadding")
    String padding;

    @Option(hidden = true, names = { "-q", "--quiet" }, defaultValue = "false")
    boolean quiet;
```

they are hidden, that and the fact it is not documented says it all, it is not supported ATM IMHO

Comment (michalvavrik) [reply QUARKUS-3456.md:54]: https://issues.redhat.com/browse/QUARKUS-3456?focusedId=25161121&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25161121

**Review by rsvoboda — APPROVED** [review]
LGTM, thanks for quick update. Open questions can be reflected once we get feedback on them.

**Pull request merged**